### PR TITLE
fix: list endpoint should return linked users

### PR DESCRIPTION
**Pull request by danielgent**
I wonder how much slower this will make the endpoint... As discussed, we want to do this in SQL to not have so many fetches _BUT_ the public won't be using this endpoint

### CODE REVIEW

File: test/integration/management-api/users.spec.ts [hunk 1]
```diff
@@ -6,31 +6,6 @@ import { getEnv } from "../helpers/test-client";
 import createTestUsers from "../helpers/createTestUsers";
 
 describe("users management API endpoint", () => {
-  describe("GET", () => {
```
Comment (danielgent) [inline test/integration/management-api/users.spec.ts:9]: I moved this down with the other tests that query the `/` `GET` endpoint

**Comment by danielgent** [comment]
Let me know what you think of this @markusahlstrand . It's not perfect but it's in the direction of what Auth0 does

**Review by markusahlstrand — APPROVED** [review]
Looks good but think a join would be much faster. A small improvement would be to make one query first and then fetch a list of users in a single call (which we might not support yet..)

**Comment by danielgent** [comment]
> Looks good but think a join would be much faster. A small improvement would be to make one query first and then fetch a list of users in a single call (which we might not support yet..)

indeed! We've discussed this a bit and it would be worth it *BUT* I'd be interested in seeing how slow it is first :sunglasses: 

We've got actually functionality to build before doing speed improvements :smile:

**Pull request merged**